runtime/parachains/src/scheduler.rs

```diff
@@ -550,7 +553,12 @@ impl<T: Trait> Module<T> {
 
 		let session_start = <SessionStartBlock<T>>::get();
 		let blocks_since_session_start = now.saturating_sub(session_start);
-		let blocks_since_last_rotation = blocks_since_session_start % config.parachain_rotation_frequency;
+		let no_rotation = config.group_rotation_frequency.is_zero();
+		let blocks_since_last_rotation = if no_rotation {
+			<T::BlockNumber>::zero()
```

Comment:
we know that `chain_availability_period` and `thread_availability_period` is at least 1
https://github.com/paritytech/polkadot/blob/b481d6c07902a2dd0e867411c63442035525610d/runtime/parachains/src/configuration.rs#L54-L58
so setting `blocks_since_last_rotation` to zero will not take the cutoff branches